### PR TITLE
BoolLiteral and Boolean object

### PR DESCRIPTION
**Pull request by andrzejkupczyk**
Created new objects for boolean values (suggested by @jgoux in https://github.com/nicolopignatelli/valueobjects/issues/13). According to PHP documentation about the [boolean type](http://php.net/manual/en/language.types.boolean.php) and the [filter for boolean validation](http://php.net/manual/en/filter.filters.validate.php), there should be two of them:  
- `BoolLiteral` which represents a PHP native bool (`TRUE`, `FALSE`)
- `Boolean` for strings representing _boolean strings_, e.g. `"true"`, `"off"`, `"yes"`

I'm open to any suggestions about this approach.

### CODE REVIEW

File: tests/BoolLiteral/BooleanValuesProvider.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace ValueObjects\Tests\BoolLiteral;
+
+trait BooleanValuesProvider
```
Comment (nicolopignarocket) [inline tests/BoolLiteral/BooleanValuesProvider.php]: define a trait to just provide two test the same data provider seems really overkill to me

File: src/BoolLiteral/BoolLiteral.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    /**
+     * Returns a string representation of the object
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return \strval($this->value);
```
Comment (nicolopignarocket) [inline src/BoolLiteral/BoolLiteral.php]: I'd expect this method to return either "true" or "false", not "1" or ""

File: tests/BoolLiteral/BooleanValuesProvider.php (hunk 1 above)
Comment (andrzejkupczyk) [inline tests/BoolLiteral/BooleanValuesProvider.php]: I agree. Transformed this trait into the `BooleanTestCase` class.

File: src/BoolLiteral/BoolLiteral.php (hunk 2 above)
Comment (andrzejkupczyk) [inline src/BoolLiteral/BoolLiteral.php]: Me too, you're absolutely right.

**Comment by nicolopignatelli** [comment]
I was also thinking that we should rename things a bit before merging.
Boolean should be renamed in BooleanString, since it exactly represent "a string with a boolean meaning". Also the namespace should be probably rename to the more generic name "Boolean"

**Comment by andrzejkupczyk** [comment]
You're right. I was thinking about that before creating pull request.  
Renamed the Boolean object and the namespace.

**Comment by nicolopignatelli** [comment]
thank you

**Pull request merged**